eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/ExplorerApi.scala

```diff
@@ -181,21 +199,37 @@ object ExplorerApi {
   }
 
   /** Query https://blockstream.info/ to fetch block headers. */
-  case class BlockstreamExplorer() extends Esplora {
+  case class BlockstreamExplorer(socksProxy_opt: Option[Socks5ProxyParams]) extends Esplora {
     override val name = "blockstream.info"
-    override val baseUris = Map(
-      Block.TestnetGenesisBlock.hash -> uri"https://blockstream.info/testnet/api",
-      Block.LivenetGenesisBlock.hash -> uri"https://blockstream.info/api"
-    )
+    override val baseUris = socksProxy_opt match {
+      case Some(_) =>
+        Map(
+          Block.TestnetGenesisBlock.hash -> uri"http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/testnet/api",
+          Block.LivenetGenesisBlock.hash -> uri"http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/api"
```

Comment:
Can you share where you found these (and also the `mempool.space` ones)?
I can't find it on their websites.

Reply:
```shell
$ curl -i -x socks5://localhost:9050 https://blockstream.info
HTTP/2 200 
server: nginx
...
onion-location: http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/
```

```shell
$ curl -i -x socks5://localhost:9050 https://mempool.space
HTTP/2 200 
server: nginx
...
onion-location: http://mempoolhqx4isw62xs7abwphsq7ldayuidyx2v2oethdhhj6mlo2r6ad.onion/
```